spec/ruby_llm/tool_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'ruby_llm/tool'
+
+RSpec.describe RubyLLM::Tool do
+  describe '#name' do
+    it 'converts class name to snake_case and removes _tool suffix' do
+      class SampleTool < RubyLLM::Tool; end
+      expect(SampleTool.new.name).to eq('sample')
+    end
+
+    it 'replaces unsupported characters with -' do
+      class SampleTòol < RubyLLM::Tool; end
+      expect(SampleTòol.new.name).to eq('sample_t-ol')
+    end
```

Comment:
This becomes unnecessary because of the next test

Reply:
true!